app/controllers/api/vcs.php

```diff
@@ -721,39 +818,86 @@
+
+                if ($type === 'framework') {
+                    $frameworkDetector = new Framework($files, $packager);
+                    $frameworkDetector
+                        ->addOption(new Flutter())
+                        ->addOption(new Nuxt())
+                        ->addOption(new Astro())
+                        ->addOption(new Remix())
+                        ->addOption(new SvelteKit())
+                        ->addOption(new NextJs());
+
+                    $detectedFramework = $frameworkDetector->detect();
+
+                    if (!\is_null($detectedFramework)) {
+                        $framework = $detectedFramework->getName();
+                    } else {
+                        $framework = 'other';
+                    }
+
+                    $frameworks = Config::getParam('frameworks');
+                    if (!\in_array($framework, array_keys($frameworks), true)) {
+                        $framework = 'other';
+                    }
+                    $repo['framework'] = $framework;
+                } else {
                     $languages = $github->listRepositoryLanguages($repo['organization'], $repo['name']);
 
-                    $detectorFactory = new Detector($files, $languages);
-
-                    $detectorFactory
-                        ->addDetector(new JavaScript())
-                        ->addDetector(new Bun())
-                        ->addDetector(new PHP())
-                        ->addDetector(new Python())
-                        ->addDetector(new Dart())
-                        ->addDetector(new Swift())
-                        ->addDetector(new Ruby())
-                        ->addDetector(new Java())
-                        ->addDetector(new CPP())
-                        ->addDetector(new Deno())
-                        ->addDetector(new Dotnet());
-
-                    $runtime = $detectorFactory->detect();
-
-                    $runtimes = Config::getParam('runtimes');
-                    $runtimeDetail = \array_reverse(\array_filter(\array_keys($runtimes), function ($key) use ($runtime, $runtimes) {
-                        return $runtimes[$key]['key'] === $runtime;
-                    }))[0] ?? '';
-
-                    $repo['runtime'] = $runtimeDetail;
-                } catch (Throwable $error) {
-                    $repo['runtime'] = "";
-                    Console::warning("Runtime not detected for " . $repo['organization'] . "/" . $repo['name']);
+                    $strategies = [
+                        new Strategy(Strategy::FILEMATCH),
+                        new Strategy(Strategy::LANGUAGES),
+                        new Strategy(Strategy::EXTENSION),
+                    ];
+
+                    foreach ($strategies as $strategy) {
+                        $redector = new Runtime($strategy === Strategy::LANGUAGES ? $languages : $files, $strategy, $packager);
```

Comment:
redector?

Reply:
Detector. Welp, AI would not make that mistake 🙈